packages/dx-react-scheduler-material-ui/src/templates/appointment-form/recurrence/radio-group/end-repeat-editor.jsx

```diff
@@ -0,0 +1,205 @@
+import Grid from '@material-ui/core/Grid';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import moment from 'moment';
+import { NUMBER_EDITOR, getRecurrenceOptions, changeRecurrenceOptions } from '@devexpress/dx-scheduler-core';
+
+const styles = ({ spacing, typography }) => ({
+  textEditor: {
+    width: 'calc(100% - 4.5em)',
+  },
+  label: {
+    width: '4.5em',
+  },
+  input: {
+    paddingBottom: spacing(2.75),
+  },
+  radioLabel: {
+    fontSize: typography.fontSize + 1,
+  },
+  dateEditor: {
+    width: 'calc(100% - 4.5em)',
+  },
+  formControl: {
+    marginRight: 0,
+  },
+  controlLabel: {
+    width: '100%',
+  },
+});
+
+const EndRepeatEditorBase = ({
+  classes,
+  getMessage,
+  labelComponent: Label,
+  textEditorComponent: TextEditor,
+  dateEditorComponent: DateEditor,
+  onFieldChange,
+  appointmentData,
+  locale,
+  readOnly,
+  ...restProps
+}) => {
+  const [count, setCount] = React.useState(1);
+  const [endDate, setEndDate] = React.useState(appointmentData.endDate);
+
+  const { rRule } = appointmentData;
+  const recurrenceOptions = React.useMemo(() => getRecurrenceOptions(rRule) || {}, [rRule]);
+  const changeRecurrenceCount = React.useCallback(nextCount => onFieldChange({
+    rRule: changeRecurrenceOptions({ ...recurrenceOptions, count: nextCount }),
+  }), [recurrenceOptions, onFieldChange]);
+  const changeRecurrenceEndDate = React.useCallback((date) => {
+    if (moment(date).isValid()) {
+      onFieldChange({
+        rRule: changeRecurrenceOptions({ ...recurrenceOptions, until: date }),
+      });
+    }
+  }, [recurrenceOptions, onFieldChange]);
+
+  const countEditorProps = React.useMemo(() => ({
+    endAdornment: <InputAdornment position="end">{getMessage('occurrencesLabel')}</InputAdornment>,
+  }), []);
+
+  const recurrenceCount = recurrenceOptions.count || count;
+  const recurrenceEndDate = recurrenceOptions.until || endDate;
+  let value;
+  if (recurrenceOptions.count) {
+    value = 'endAfter';
+  } else if (recurrenceOptions.until) {
+    value = 'endBy';
+  } else value = 'never';
+
+  const onRadioGroupValueChange = (event) => {
+    switch (event.target.value) {
+      case 'endAfter':
+        setEndDate(recurrenceOptions.until || endDate);
+        onFieldChange({
+          rRule: changeRecurrenceOptions({
+            ...recurrenceOptions, count, until: undefined,
+          }),
+        });
+        break;
+      case 'endBy':
+        setCount(recurrenceOptions.count || count);
+        onFieldChange({
+          rRule: changeRecurrenceOptions({
+            ...recurrenceOptions, count: undefined, until: endDate,
+          }),
+        });
+        break;
+      case 'never':
+        setEndDate(recurrenceOptions.until || endDate);
+        setCount(recurrenceOptions.count || count);
+        onFieldChange({
+          rRule: changeRecurrenceOptions({
+            ...recurrenceOptions, count: undefined, until: undefined,
```

Comment:
We can for simplicity only initialize change = { count, until } in this switch and apply it below